python/tvm/relay/op/contrib/cmsisnn.py

```diff
@@ -122,6 +130,25 @@ def check_qnn_conv2d(pattern):
         conv2d_input = conv2d.args[0]
         conv2d_weight = conv2d.args[1]
 
+        # check if sum of paddings from pad() and conv2d() satisfies CMSIS-NN constraints
+        can_pad_be_fused = True
+        if isinstance(conv2d_input, tvm.relay.expr.Call) and str(conv2d_input.op.name) == "nn.pad":
+            data_layout = conv2d.attrs.data_layout
+            conv2d_padding = conv2d.attrs.padding  # (top, left, bottom, right)
+            pad = conv2d_input
+            pad_width = pad.attrs.pad_width
+            pad_width_w = pad_width[data_layout.find("W")]  # (left, right)
+            pad_width_h = pad_width[data_layout.find("H")]  # (top, bottom)
+            # calculate effective padding post pad fusion
+            pad_top = pad_width_h[0] + conv2d_padding[0]
+            pad_bottom = pad_width_h[1] + conv2d_padding[2]
+            pad_left = pad_width_w[0] + conv2d_padding[1]
+            pad_right = pad_width_w[1] + conv2d_padding[3]
+            # check if difference in the side paddings is 1 along each dimension
+            pad_w_diff = int(pad_right - pad_left)
```

Comment:
Why is this cast to int necessary? What are the data types of pad_right and pad_left?

Reply:
They are IntImm. When they are used in conditional checks, it returns IntImm in place of boolean which is what we'd expect normally. We have seen this in the past too: https://github.com/apache/tvm/blob/22ba659438a317ca59c8201430c662f86e2550fd/python/tvm/relay/op/contrib/cmsisnn.py#L134

Reply:
> They are IntImm. When they are used in conditional checks, it returns IntImm in place of boolean which is what we'd expect normally. We have seen this in the past too:
> 
> https://github.com/apache/tvm/blob/22ba659438a317ca59c8201430c662f86e2550fd/python/tvm/relay/op/contrib/cmsisnn.py#L134

Thanks for the explanation!

Reply:
No worries :+1: